src/components/select/select.js

```diff
@@ -776,6 +777,14 @@ function OptionDirective($mdButtonInkRipple, $mdUtil) {
     var optionCtrl = ctrls[0];
     var selectCtrl = ctrls[1];
 
+    if (selectCtrl.isMultiple) {
+      element.attr('md-checkbox-enabled', '');
+      if (!checkboxElement) {
+        checkboxElement = angular.element('<div class="_md-container"><div class="_md-icon"></div></div>');
```

Comment:
This isn't there yet; you're still performing the html -> dom operation for _every single item, everywhere_. The goal is to only do that operation once _ever_, and then clone the result for each item.

You want to do something like 
```js
var CHECKBOX_SELECTION_INDICATOR = 
    angular.element('<div class="_md-container"><div class="_md-icon"></div></div>');
```
right inside of `OptionDirective` before the `return`, and then clone that.